javatests/arcs/core/data/proto/SchemaProtoDecoderTest.kt

```diff
@@ -52,4 +57,160 @@ class SchemaProtoDecoderTest {
         assertThat(schema.fields.singletons).isEqualTo(
             mapOf("text" to FieldType.Text, "bool" to FieldType.Boolean))
     }
+
+    @Test
+    fun decodesCollectionsInSchemaProto() {
+        // schema
+        //   text: [ Text ]
+        //   bool: [ Boolean ]
+        val schemaProtoText = """
+        fields {
+          key: "text"
+          value: {
+            collection {
+              collection_type: {
+                primitive: TEXT  
+              } 
+            }
+          }
+        }
+        fields {
+          key: "bool"
+          value: {
+            collection {
+              collection_type: {
+                primitive: BOOLEAN
+              } 
+            }
+          }
+        }
```

Comment:
Consider writing in a more compact form? I think this could be more readable, especially for the larger examples further down:

```
fields {
  key: "text"
  value: {
    collection { collection_type: { primitive: TEXT } }
  }
}
fields {
  key: "bool"
  value: {
    collection { collection_type: { primitive: BOOLEAN } }
  }
}
```